sklearn/neighbors/_nearest_centroid.py

```diff
@@ -255,7 +314,100 @@ def predict(self, X):
         """
         check_is_fitted(self)
 
-        X = self._validate_data(X, accept_sparse="csr", reset=False)
-        return self.classes_[
-            pairwise_distances_argmin(X, self.centroids_, metric=self.metric)
-        ]
+        if len(np.unique(self.class_priors_, return_counts=True)[0]) == 1:
```

Comment:
This does not take into account the possibility that users manually input uniform class priors, e.g. `[0.5, 0.5]`.

Regardless, do you think this is the best way to check that `self.class_priors_` is uniform?

Reply:
I would not bother about the case of passing user-defined case since it would be cover by the "uniform" case.

Reply:
It's not covered as shown here:

```
        if self.priors == "empirical":  # estimate priors from sample
            _, class_counts = np.unique(y, return_inverse=True)  # non-negative ints
            self.class_priors_ = np.bincount(class_counts) / float(len(y))
        elif self.priors == "uniform":
            self.class_priors_ = np.asarray([1 / n_classes] * n_classes)
        else:
            self.class_priors_ = np.asarray(self.priors)
```

Reply:
I would prefer the following since we expect to have a sum to 1:

```suggestion
        if np.isclose(self.class_priors_, 1 / len(self.classes_)).all():
```